crates/uv-workspace/src/build_backend.rs

```diff
@@ -0,0 +1,168 @@
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+use uv_macros::OptionsMetadata;
+use uv_pypi_types::Identifier;
+
+/// Settings for the uv build backend (`uv_build`).
+///
+/// !!! note
+///
+///     The uv build backend is currently in preview and may change in any future release.
+///
+/// Note that those settings only apply when using the `uv_build` backend, other build backends
+/// (such as hatchling) have their own configuration.
+#[derive(Deserialize, Serialize, OptionsMetadata, Debug, Clone, PartialEq, Eq)]
+#[serde(default, rename_all = "kebab-case")]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub struct BuildBackendSettings {
+    /// The directory that contains the module directory, usually `src`, or an empty path when
+    /// using the flat layout over the src layout.
+    #[option(
+        default = r#""src""#,
+        value_type = "str",
+        example = r#"module-root = """#
+    )]
+    pub module_root: PathBuf,
+
+    /// The name of the module directory inside `module-root`.
+    ///
+    /// The default module name is the package name with dots and dashes replaced by underscores.
+    ///
+    /// Note that using this option runs the risk of creating two packages with different names but
+    /// the same module names. Installing such packages together leads to unspecified behavior,
+    /// often with corrupted files or directory trees.
```

Comment:
We had bug reports caused by this exact problem, but unfortunately it's a somewhat popular feature (scikit-learn -> sklearn, Pillow -> PIL, etc.). It also allows user to pick their preferred package name even if it is already taken on PyPI by only renaming the installation name to something free while docs and `import <...>` can have any name.